### DISCUSSION TITLE
Possible with this package?

### DISCUSSION THREAD

**Discussion by HeyItsRyanJames** [gdelataillade/alarm#310] [Q&A]
Hey there @gdelataillade,

Thanks for this great package! I think it will be useful for what I'm trying to do. I'm building an app and want users to be able to set an alarm. The alarm sound, however, would be an audio recording stored on a server somewhere, and this recording would be unique to a given user. No two users will have the same recording. The app would fetch the recording from the server and ideally play as the user's alarm when the time comes. Is this possible?

Additionally, I understand that iOS is quite restrictive when it comes to alarm functionality from 3P app developers. I'm wondering what behavior I can expect if the app is killed. Your table says "Notification" is possible, but will that notification have any sound at all? 

Thank you!

**Comment by gdelataillade** [comment]
Hi @HeyItsRyanJames,

Thank you for your interest in the package! I’m glad you find it useful for your project. Here’s some information to address your questions:

1) **Custom Audio for Alarms**: Yes, you can use a custom audio recording fetched from your server as the alarm sound. However, it’s better to download the audio file beforehand and provide its local file path to the alarm configuration, rather than attempting to play it directly from the URL when the alarm triggers. This approach avoids two potential issues:
	•	If the device has no internet connection at the time the alarm triggers, the audio won’t be downloaded.
	•	If the audio file is large or takes time to buffer, it could delay the alarm.
2) **iOS Alarm Behavior**: iOS is indeed very restrictive regarding alarm functionality for third-party apps. Specifically:
	•	Alarm notifications on iOS will not include any sound. This is intentional to ensure the notification sound doesn’t interfere with the custom alarm audio.
	•	To mitigate app-kill scenarios, the Alarm.setWarningNotificationOnKill(title, body) method is available. It shows a notification warning the user that the app has been killed and their alarms won’t ring unless they reopen the app to reschedule them. This limitation applies only to iOS—on Android, alarms continue to work even if the app is killed.

Feel free to reach out if you have any more questions or need further assistance!
